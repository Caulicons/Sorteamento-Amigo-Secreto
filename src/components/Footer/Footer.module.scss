@import 'styles/variaveis';

.footer {
  display: flex;
  flex-direction: column;
  gap: 27px;
  justify-content: center;
  align-items: center;
  &_button {
    width: 180px;
    padding:16px; 
    background-color: $laranja;
    color: $branco;
    font-weight: 600;
    font-size: 16px;
    border-radius: 2rem;
    border: solid 1px #000;
    box-shadow: 4px 4px 0px #000000;
  &:hover {
    cursor: pointer;
    background-color: $azul;
  }
  }

  p{
    text-align: center;
    align-items: center;
  }
}

@media screen and (min-width: $width-desktop) {
  .footer {
    flex-direction: row;

    &_button {
      width: 250px;
      padding:30px 32px; 
    }
  }
}