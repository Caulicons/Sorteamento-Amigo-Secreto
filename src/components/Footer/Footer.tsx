import { useNavigate } from "react-router-dom";
import useListaDeParticipantes from "state/hooks/useListaDeParticipantes";
import sacolas from 'assets/sacolas.png';
import styles from './Footer.module.scss'
import { useState } from "react";
import { useSorteador } from "state/hooks/useSorteador";

const Footer = () => {
  const [poucosParticipantes, setPoucosParticipantes] = useState<boolean>(false);
  const participantens = useListaDeParticipantes();

  const irParaOSorteio = useNavigate();
   const sortear = useSorteador(); 

  const iniciarBrincadeira = () => {
    
    if(participantens.length < 3){

      setPoucosParticipantes(true)
      setTimeout( () => setPoucosParticipantes(false), 3000)
      return
    }
    sortear() 
    irParaOSorteio('/Sorteio');
  }

  return (
    <footer className={styles.footer}> 
      <button onClick={iniciarBrincadeira} className={styles.footer_button}>
        Iniciar Brincadeira !
      </button>
      {poucosParticipantes && <p role='alert'>O sorteio deve ter no m√≠nimo 3 participantes.</p>}
      <img src={sacolas} alt="Sacola de presentes." className={styles.footer_sacolas}/>
    </footer>
  );
}

export default Footer;