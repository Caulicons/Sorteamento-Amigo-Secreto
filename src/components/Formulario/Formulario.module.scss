@import 'styles/variaveis';

.form {
  display: flex;
  flex-direction: column; 
  align-items: center;
  width: 100%;
  row-gap: 9px;
  &_input {
    padding: 15px 16px;
    border-radius: 2rem;
    border: solid 2px #000; 
    box-shadow: 3px 3px;
    width: 80%;
    font-size: 14px;
  }

  &_button {
    width: 180px;
    padding: 12px 8px;
    border-radius: 2rem;
    box-shadow: 3px 3px;
    background-color: $cinza;
  }
   
}

.mensagemError {
  padding: 18px 24px;
  background-color: rgb(255, 171, 171);
  border: solid 1px rgb(156, 0, 0);
  color: rgb(128, 0, 0); 
  border-radius: 10px;
}

@media screen and (min-width: $width-desktop) {
  .form {
    flex-direction: row; 
    width: 70%;
    max-width: 800px;
    &_input {
      border-radius: 37.5px 0px 0px 37.5px;
      border: solid 2px #000; 
      box-shadow: 3px 3px;
      width: 75%;
      padding: 24px;
      font-size: 17px;
    }
  
    &_button {
      width: 180px;
      padding: 24px;
      border-radius: 0px 37.5px 37.5px 0px;
      box-shadow: 4px 4px;

    }
  }
}
