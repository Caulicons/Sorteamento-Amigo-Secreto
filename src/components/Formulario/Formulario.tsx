import { useRef, useState } from "react";
import { useAdicionarParticipantes } from "state/hooks/useAdicionarParticipante";
import { useMensagemDeError } from "state/hooks/useMensagemDeError";
import styles from './Formulario.module.scss';

const Formulario = () => {
  const [nome, setNome] = useState<string>('');
  const adicionarNaLista = useAdicionarParticipantes();
  const inputRef = useRef<HTMLInputElement>(null)

  const mensagemError = useMensagemDeError()

  const adicionarParticipante = (evento: React.FormEvent<HTMLFormElement>) => {
    evento.preventDefault();

    if (nome) {
      adicionarNaLista(nome as string)
      setNome('');
      inputRef.current?.focus()
    }
  }

  return (
   <><form onSubmit={adicionarParticipante} className={styles.form}>
      <input
        className={styles.form_input}
        ref={inputRef}
        type="text"
        placeholder="Insira os nomes dos participantes"
        value={nome}
        onChange={event => setNome(event.target.value)}
      />
      <button type="submit" className={styles.form_button}>Adicionar</button>

    </form>
    {mensagemError && <p role='alert' className={styles.mensagemError}>{mensagemError}</p>}
    </>
  )
}

export default Formulario;