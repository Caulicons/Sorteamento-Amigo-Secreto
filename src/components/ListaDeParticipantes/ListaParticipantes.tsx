import useListaDeParticipantes from "state/hooks/useListaDeParticipantes";
import styles from './ListaParticipantes.module.scss';


const ListaParticipantes = () => {

  const participantes: string[] = useListaDeParticipantes()

  return (
    <ul className={styles.participantes}>
      {participantes.map( participante => <li className={styles.participante} key={participante}>{participante}</li> )}
    </ul>
  )
};  

export default ListaParticipantes;