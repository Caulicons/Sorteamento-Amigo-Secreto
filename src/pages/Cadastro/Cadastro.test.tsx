import { render } from "@testing-library/react"
import Footer from "components/Footer/Footer"
import Formulario from "components/Formulario/Formulario"
import ListaParticipantes from "components/ListaDeParticipantes/ListaParticipantes"
import { RecoilRoot } from "recoil"

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate
  }
})

describe('O componente Cadastro deve conter essa estrutura', () => {

  test('Tirar um print', () => {

    const { container } = render(<RecoilRoot>
      <Formulario />
      <ListaParticipantes />
      <Footer />
    </RecoilRoot>)

    expect(container).toMatchSnapshot()
  })
})