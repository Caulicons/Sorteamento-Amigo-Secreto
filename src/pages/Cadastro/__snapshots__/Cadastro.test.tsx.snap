// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`O componente Cadastro deve conter essa estrutura Tirar um print 1`] = `
<div>
  <form
    class="form"
  >
    <input
      class="form_input"
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      class="form_button"
      type="submit"
    >
      Adicionar
    </button>
  </form>
  
  <ul
    class="participantes"
  />
  <footer
    class="footer"
  >
    <button
      class="footer_button"
    >
      Iniciar Brincadeira !
    </button>
    <img
      alt="Sacola de presentes."
      class="footer_sacolas"
      src="sacolas.png"
    />
  </footer>
</div>
`;
